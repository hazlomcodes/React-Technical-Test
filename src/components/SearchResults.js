import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {

  if (!results?.length) {
    return <p>No results</p>;
  } else {
    return (
      <div className="search-results">
        {results.map((image, index) => {
          return (
            <img
              className="search-results__image"
              src={image}
              alt="space_image"
              key={index}
            />
          );
        }
       )}
        </div>
    );
  }
}; 

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
};

  
export default SearchResults;
