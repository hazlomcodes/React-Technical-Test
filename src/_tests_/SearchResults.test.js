import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search results", () => {
  const { asFragment } = render(<SearchResults />);

  it("render correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  })
})

it("renders `No results` if there is no result", () => {
  const validProps = [];
  render(<SearchResults results={validProps} />);
  const placeholderMessage = screen.getByText("No results");
  
  expect(placeholderMessage).toBeInTheDocument();
})