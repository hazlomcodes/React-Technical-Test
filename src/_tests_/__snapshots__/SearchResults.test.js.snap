// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results render correctly 1`] = `
<DocumentFragment>
  <p>
    No results
  </p>
</DocumentFragment>
`;
